### PR TITLE
ENH: allow pyarrow.Table in args to write_table

### PR DESCRIPTION
**Pull request by genematx**
This allows SQLAdapter to accept entire pyarrow Table as arguments to `write_table` and `append_partition`, which removes the need to initialize intermediary pandas DataFrames.

### Checklist
- [x] Add a Changelog entry
- [ ] ~~Add the ticket number which this PR closes to the comment section~~

### CODE REVIEW

File: tiled/adapters/sql.py [hunk 1]
```diff
@@ -238,27 +238,28 @@ def items(self) -> Iterator[Tuple[str, ArrayAdapter]]:
...
         partition: int,
     ) -> None:
-        """
-        "Function to write the data as arrow format."
+        """Write the data as arrow format
 
         Parameters
         ----------
-        data : data to append into the database. Can be a list of record batch, or pandas dataframe.
+        data : data to append to the database. Can be a list of record batches, pyarrow table, or pandas dataframe.
```
Comment (skarakuzu) [inline tiled/adapters/sql.py:248]: `data : data to append to the database. Can be a record_batch, a list of record batches, pyarrow table, or pandas dataframe.`

File: tiled/adapters/sql.py [hunk 2]
```diff
@@ -238,27 +238,28 @@ def items(self) -> Iterator[Tuple[str, ArrayAdapter]]:
...
         partition : the partition index to write.
-        Returns
-        -------
         """
-        if isinstance(data, pandas.DataFrame):
+        # Conver the data to pyarrow table
```
Comment (skarakuzu) [inline tiled/adapters/sql.py:251]: `# Convert the data to pyarrow table`

**Review by skarakuzu — DISMISSED** [review]

File: tiled/adapters/sql.py (hunk 1 above)
Comment (genematx) [reply tiled/adapters/sql.py:248]: You can use the "Add suggestion" button, which automatically creates a template for a commit.

File: tiled/adapters/sql.py
[the remark this answers is not in the document]
Comment (genematx) [reply tiled/adapters/sql.py:1]: What edge cases? It's just a conversion to `pyarrow.Table` outside of the `append_partition` method; essentially, what you do in lines 254-261 already.
Added a basic test for `append_partition` from a single table; the rest of datatypes seem to be already covered in other tests.
